types/config.go

```diff
@@ -0,0 +1,61 @@
+package types
+
+import (
+	"context"
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+	"gorm.io/gorm/schema"
+)
+
+// +kubebuilder:object:generate=true
+// ConfigQuery is used to look up and associate
+// config items with a component.
+type ConfigQuery struct {
+	ID         []string          `json:"id,omitempty"`
+	Type       string            `json:"type,omitempty"`
+	Class      string            `json:"class,omitempty"`
+	ExternalID string            `json:"external_id,omitempty"`
+	Name       string            `json:"name,omitempty"`
+	Namespace  string            `json:"namespace,omitempty"`
+	Tags       map[string]string `json:"tags,omitempty"`
+}
+
+func (c ConfigQuery) String() string {
+	s := c.Type
+	if c.Namespace != "" {
+		s += "/" + c.Namespace
+	}
+
+	if c.Name != "" {
+		s += "/" + c.Name
+	}
+
+	if len(c.Tags) > 0 {
+		s += " " + fmt.Sprintf("%v", c.Tags)
+	}
+
+	return s
+}
```

Comment:
Lets just use `%+v` or something like that here ?

Reply:
interestingly, %+v for map makes no difference.


```
fmt.Printf("%v\n", x)
fmt.Printf("%+v\n", x)
fmt.Printf("%#v\n", x)

map[address:global name:flanksource]
map[address:global name:flanksource]
map[string]string{"address":"global", "name":"flanksource"}
```

I think just `%v` looks fine.

Reply:
Lets just `%+v` or `%v` for the entire String method, since we'll need all the attributes when printing it in logs